devops/scripts/deploy-public-upstream.sh

```diff
@@ -65,7 +65,24 @@ for commit in "${commits[@]}"; do
   fi
   echo "Adding $commit:"
   git --no-pager log --format=%B -n 1 "$commit"
-  git cherry-pick -rn -X theirs "$commit" 2>&1
+  if ! git cherry-pick -rn -X theirs -m 1 "$commit"; then
+    echo "Conflict detected in $commit. Checking for deleted files."
+    conflicted_files=$(git diff --name-only --diff-filter=U)
+    for file in $conflicted_files; do
+      if ! git ls-tree -r "$commit" --name-only | grep -q "^$file$"; then
+        echo "File $file was deleted in the cherry-picked commit. Resolving by keeping the deletion."
+        git rm "$file"
+      else
+        echo "Conflict required manual resolution for $file."
```

Comment:
does this line need to bail (or set a flag to bail after)?

Reply:
It should bail on its own. When I was running it locally it caused the script to halt, which in Circle should cause the workflow to fail. I don't think we need a more explicit bail here.